ocs/ocp.py

```diff
@@ -102,22 +104,34 @@ def create(self, yaml_file, out_yaml_format=True):
 
         return self.exec_oc_cmd(command)
 
-    def delete(self, yaml_file, wait=True):
+    def delete(self, yaml_file=None, resource_name='', wait=True):
         """
         Deletes a resource
 
         Args:
             yaml_file (str): Path to a yaml file to use in 'oc delete -f
                 file.yaml
+            resource_name (str): Name of the resource you want to delete
             wait (bool): Determines if the delete command should wait to
                 completion
 
         Returns:
             Munch Obj: this object represents a returned yaml file
         """
-        command = f"delete -f {yaml_file}"
+        if yaml_file is None and not resource_name:
```

Comment:
```python
if not (yaml_file or resource_name):
```
this will do the same I think

Reply:
yep, thanks:
```python
>>> if yaml is None and not res:
...     print "123"
...     
... 
123
>>> if not (yaml and res):
...     print "123"
...     
... 
123

```

Reply:
I see in you example is `and` , but I think that you would like to do the operation if none of both yaml and res is not defined right? 
So for this example I see:
![nor image](https://www.dyclassroom.com/image/topic/logic-gate/nand-nor/nor-table.png)

Maybe I am missing something, can you elaborate please?

Reply:
yeah this example is not valid.
In the code is 'or'